libshaderc/include/shaderc.h

```diff
@@ -24,12 +24,26 @@ extern "C" {
 #include <stdint.h>
 
 typedef enum {
-  shaderc_glsl_vertex_shader,
-  shaderc_glsl_fragment_shader,
-  shaderc_glsl_compute_shader,
-  shaderc_glsl_geometry_shader,
-  shaderc_glsl_tess_control_shader,
-  shaderc_glsl_tess_evaluation_shader
+  // Forced shader kinds. These shader kinds force the compiler to compile the
+  // source code as the specified kind of shader.
+  shaderc_glsl_forced_vertex_shader,
```

Comment:
But then how does the compiler know the shader kind, if the arg is set to shaderc_infer_from_source (or shaderc_glsl_deduce_from_pragma) and #pragma is not found?
We won't have any hints about which shader kind to use in this case, unless we make the deducer callback exposed to client again. Libshaderc doesn't do the job to infer the shader kind from file name, I think.